.github/workflows/docs.yml

```diff
@@ -21,7 +21,7 @@ jobs:
       - name: "Check for changes in documentation"
         id: check_doc_changes
         run: |
-          if test -z "$(git diff --name-only HEAD~1..HEAD -- doc/)";
+          if test -z "$(git diff --name-only HEAD~1..HEAD -- doc/ .github/workflows/docs.yml)";
           then
             echo "No changes in doc/"
             echo "diff=false" >> $GITHUB_OUTPUT
```

Comment:
Orthogonal to this PR, but: if I read this action right, if `diff=false` then all subsequent steps are skipped *except for the installation of TeXLive -- is that intentional?

Or do you actually want to skip *all* steps? In that case I would recommend to replace this code by a workflow activation filter based on modified paths... That is, instead of
```
on:
  push:
    branches: [trunk]
```
write something like
```
on:
  push:
    branches:
      - trunk
    paths:
      - 'doc/**'
      - '.github/workflows/docs.yml'
```
then this workflow will only be triggered by pushes that modify one of those paths. In particular it can then avoid spinning up a VM, conserving resources. For details, [here is the documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore)

Reply:
Ah that is much better than the existing approach, thanks.
I will switch to using that